dev/tools/pin-ci.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# Use this script to pin the commit used by the developments tracked by the CI
+
+OVERLAYS="./dev/ci/ci-basic-overlay.sh"
+
+process_development() {
+  local DEV=$1
+  local REPO_VAR="${DEV}_CI_GITURL"
+  local REPO=${!REPO_VAR}
+  local BRANCH_VAR="${DEV}_CI_REF"
+  local BRANCH=${!BRANCH_VAR}
+  if [[ -z "$BRANCH" ]]
+  then
+    echo "$DEV has no branch set, skipping"
+    return 0
+  fi
+  if [[ $BRANCH =~ ^[a-f0-9]{40}$ ]]
+  then
+    echo "$DEV is already set to hash $BRANCH, skipping"
+    return 0
+  fi
+  echo "Resolving $DEV as $BRANCH from $REPO"
+  local HASH=$(git ls-remote --heads $REPO $BRANCH | cut -f 1)
+  if [[ -z "$HASH" ]]
+  then
+    echo "Could not resolve reference $BRANCH for $DEV (something went wrong), skipping"
+    return 0
+  fi
+  read -p "Expand $DEV from $BRANCH to $HASH? [y/N] " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    sed -i -e "s/$BRANCH_VAR:=$BRANCH/$BRANCH_VAR:=$HASH/" $OVERLAYS
```

Comment:
IIRC apple has issues with -i

Reply:
This is a Bash script, I am not trying to support the gazillion of POSIX-like shells around.

Reply:
It's a sed difference, not about bash.

Reply:
```suggestion
    # use -i"" to be compatible with MacOS; see, e.g., https://stackoverflow.com/a/7573438/377022
    sed -i"" -e "s/$BRANCH_VAR:=$BRANCH/$BRANCH_VAR:=$HASH/" $OVERLAYS
```

Reply:
In that link see how it says "note the space", `-i""` is the same as `-i`.

Reply:
Ah, right, thanks.  Updated.